src/CTA.Rules.Models/Actions/ExpressionAction.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace CTA.Rules.Models
+{
+    public class ExpressionAction : GenericAction
+    {
+        public Func<SyntaxGenerator, SyntaxNode, SyntaxNode> ExpressionActionFunc { get; set; }
+        public override bool Equals(object obj)
+        {
+            var action = (ExpressionAction)obj;
+            return action.Key == this.Key
+                && action.Value == this.Value
+                && action.ExpressionActionFunc.Method.Name == this.ExpressionActionFunc.Method.Name;
+        }
+
+        public override int GetHashCode()
+        {
+            return Key.GetHashCode()
+                + 3 * Value.GetHashCode()
+                + 5 * (ExpressionActionFunc != null ? ExpressionActionFunc.Method.Name.GetHashCode() : 0);
```

Comment:
Any null checks needed here?

Reply:
Good catch, the previous null checks do allow these to be null now so I'll check for that too here, thanks.